lib/examples/shell.js

```diff
@@ -0,0 +1,22 @@
+// transpile:main
+import { connectToDevice } from '../../index.js';
+import { CONNECTION_TYPES } from '../constants';
+
+async function start () {
+  let adb = connectToDevice(CONNECTION_TYPES.USB);
+  await adb.start();
+
+  console.log("device started");
+  // opening a stream into the device causes issues atm
+  // let command = {
+  //   type:   "shell"
+  // , string: "ls"
+  // };
+  // await adb.open(command);
+  // console.log("shell opened");
+  console.log("closing device!");
+  await adb.closeConnection();
+}
+
+console.log("starting");
+start();
```

Comment:
you might want to wrap this so any errors are output instead of swallowed in the promise rejection. the way that i usually do this is:

```js
import { asyncify } from 'asyncbox';
```

then

```js
asyncify(start);
```

It just handles the boilerplate of logging errors to the console if something in the promise chain is rejected